Fix for issue 18

renfeiw opened the pull request:
- added untracked test files into .gitignore
- added code for test file clean up for gctest, gcperftest, rastest and porttest

Signed-off-by: renfeiw <renfeiw@ca.ibm.com>

rservant commented on .gitignore:
```diff
@@ -123,3 +123,14 @@ port/omrsyslogmessages.h
 x64/
+
+# Test Output
+*.dylib.dbg
```
.dylib.dbg is not a test output file.  I suggest moving this up to be with the .dylib entry.

renfeiw replied:
Thanks Ron, it is updated.

alin00 commented:
Renfei: Please run an omr personal build and provide the vmfarm URL that shows these tests all pass.

@charliegracie Can you comment with your proposed git commit comment format?

mstoodle commented:
Just wanted to make it clear for any non-IBM contributor watching this space...

We do not expect external contributors to run an "omr personal build" along with the vmfarm URL. We're trying to get Travis-CI enabled at the moment so that contributors can provide the same kind of test result that Angela has asked for here (for the record, we know renfeiw and know he has the access to do these things).

Until we can get Travis-CI going ( see [here](https://github.com/eclipse/omr/issues/21) ), external contributors can simply follow the guidelines in the CONTRIBUTING.md file and we'll deal with it.

renfeiw commented:
All the tests have passed in the personal build: http://vmfarm.ottawa.ibm.com/jobs_by_category.php?category=33546&build_id=295768

charliegracie commented:
I think the commit comment is ok for now until we reach a general consensus on what the official format should be.  I might have started my commit message with something like:

Clean up test material to resolve #18

alin00 commented:
I think this pull request includes only a partial fix for #18.  Please update this request when you have the complete fix ready.

renfeiw commented:
Updated the commit comment and added clean up code in subscriberForkTest and subscriberTest.
Personal build passed at: http://vmfarm.ottawa.ibm.com/build_info.php?build_id=295787

alin00 commented on .gitignore:
```diff
@@ -123,3 +124,13 @@ port/omrsyslogmessages.h
 port/MSG00001.bin
 x64/
+
+# Test Output
```
The rastest output files aren't being ignored. Shouldn't they be consistent?

alin00 commented on fvtest/rastest/subscriberForkTest.cpp:
```diff
@@ -143,6 +143,12 @@ runTest(omrthread_entrypoint_t entrypoint)
 		postForkParent(vmthread, agent, &testVM, pipedata);
 	}
+
+	/* Clean up test output file */
```
Are these files created by the agent? If so, they should be cleaned up by the agent.

alin00 commented on fvtest/rastest/subscriberTest.cpp:
```diff
@@ -109,4 +109,10 @@ TEST(RASSubscriberTest, SubscriberAgent)
 	/*  Now clear up the VM we started for this test case. */
 	OMRTEST_ASSERT_ERROR_NONE(omrTestVMFini(&testVM));
+
+	/* Clean up test output file */
```
Same as comment on subscriberForkTest.cpp

alin00 commented on fvtest/gctest/gcTestHelpers.cpp:
```diff
@@ -60,6 +60,8 @@ GCTestEnvironment::initParams()
 				params.push_back(line);
 			}
 			omrfile_close(fileDescriptor);
+		} else if (strncmp(_argv[i], "-keepVerboseLog", strlen("-keepVerboseLog")) == 0) {
```
Why use strncmp instead of strcmp? Do you expect to accept options that have -keepVerboseLog as a prefix?

alin00 commented on fvtest/gctest/GCConfigTest.cpp:
```diff
@@ -104,10 +104,30 @@ GCConfigTest::TearDown()
 	objectTable.free();
 
-	/* close verboseManager */
+	/* close verboseManager and clean up verbose files*/
```
Please add space before */

alin00 commented on fvtest/porttest/testHelpers.cpp:
```diff
@@ -512,3 +512,23 @@ raiseSEGV(OMRPortLibrary *portLibrary, void *arg)
+			}
+			rc = omrfile_findnext(handle, resultBuffer);
+		}
+		if (handle != (uintptr_t)-1) {
```
if ((uintptr_t)-1 != handle) {

alin00 commented on fvtest/porttest/testHelpers.cpp:
```diff
@@ -512,3 +512,23 @@ raiseSEGV(OMRPortLibrary *portLibrary, void *arg)
+{
+	if (::testing::UnitTest::GetInstance()->current_test_case()->Passed()) {
+		OMRPORT_ACCESS_FROM_OMRPORT(portTestEnv->getPortLibrary());
+		uintptr_t rc, handle;
```
Declare 1 variable per line, and initialize them.

alin00 commented on fvtest/gctest/GCConfigTest.cpp:
```diff
@@ -104,10 +104,30 @@ GCConfigTest::TearDown()
+	if (false == gcTestEnv->keepLog) {
+		if (0 == numOfFiles) {
+			J9FileStat buf;
+			omrfile_stat(verboseFile, 0, &buf);
```
check return value of omrfile_stat()?  same below

renfeiw commented:
@alin00 Amended commit based on your comment. The travis ci build failed due to the issue in #29, I will revalidate my change after #28 is in.

alin00 commented:
Thanks for addressing the review comments.  I've merged #28.

renfeiw commented:
The Travis CI build passed.

alin00 commented:
Pull request for issue #18
